### PR TITLE
Update docker configs to run in docker swarm on prod

### PR DESCRIPTION
**Pull request by g3d**

### CODE REVIEW

File: config/cable.docker.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+redis: &redis
+  adapter: redis
+  url: <%= ENV.fetch("REDIS_URL") { "redis://redis:6379/1" } %>
+
+production:
+  <<: *redis
+
```
Comment (denys-medynskyi) [inline config/cable.docker.yml]: Please remove newline

File: config/database.docker.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pool:     50
+  database: pivorak_production
+  password: <%= ENV['DATABASE_PASSWORD'] %>
+
+production:
+  <<: *default
+  pool:     100
+  database: pivorak_production
+  password: <%= ENV['DATABASE_PASSWORD'] %>
+
```
Comment (denys-medynskyi) [inline config/database.docker.yml]: Please remove newline

File: docker-entrypoint.deploy.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+# Interpreter identifier
+
+set -e
+/wait && /init
+
```
Comment (denys-medynskyi) [inline docker-entrypoint.deploy.sh]: Remove newline

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -e
+
+# Then exec the container's main process (what's set as CMD in the Dockerfile).
+
+exec "$@"
+
```
Comment (denys-medynskyi) [inline docker-entrypoint.sh]: Remove newline

File: s6/etc/cont-init.d/01-name-resolving [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/execlineb -P
+
+with-contenv
+/usr/local/sbin/add_external_dns
+
```
Comment (denys-medynskyi) [inline s6/etc/cont-init.d/01-name-resolving]: Remove newline

File: s6/etc/cont-init.d/04-database [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/execlineb -P
+# Interpreter identifier
+
+with-contenv
+
+# Create db if not exist
+/usr/local/sbin/init_db
+
```
Comment (denys-medynskyi) [inline s6/etc/cont-init.d/04-database]: Remove newline here and below 👇

File: s6/etc/cont-init.d/06-whenever [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/execlineb -P
+# Interpreter identifier
+
+with-contenv
+s6-envuidgid -D 32769:32769 daemon
+# Update crontab
+/usr/local/sbin/update_cron
+
```
Comment (denys-medynskyi) [inline s6/etc/cont-init.d/06-whenever]: Do we need whenever?

File: s6/etc/cont-init.d/07-cron [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/execlineb -P
+# Interpreter identifier
+
+with-contenv
+
+# Create db if not exist
+cron
```
Comment (denys-medynskyi) [inline s6/etc/cont-init.d/07-cron]: Do we need `cron`? Is it for `FetchAllTalksDataJob` which runs daily?

**Review by denys-medynskyi — COMMENTED** [review]
1. There are many redundant newlines
2. I get an error:
```
➜  pivorak-web-app git:(feature/swarm) docker-compose build               
WARNING: Some services (api, app, postgres, redis, sidekiq) use the 'deploy' key, which will be ignored. Compose does not support 'deploy' configuration - use `docker stack deploy` to deploy to a swarm.
postgres uses an image, skipping
redis uses an image, skipping
Building app
Step 1/16 : ARG RUBY_VERSION
Step 2/16 : FROM ruby:$RUBY_VERSION
ERROR: Service 'app' failed to build: invalid reference format
```
Am I doing something wrong?
3. Am I testing it correctly?
- docker-compose build
- docker-compose run runner bundle exec rails db:setup
- docker-compose up rails
?

File: config/cable.docker.yml (hunk 1 above)
Comment (skrix) [reply config/cable.docker.yml]: it's a configuration of editor or IDE, auto adding new-line at the end of the file, please don't spam with such comment, one is enough.

File: Gemfile [hunk 9]
```diff
@@ -1,8 +1,7 @@
 # frozen_string_literal: true
-
 source 'https://rubygems.org'
 
-ruby '2.6.5'
+ruby File.open('.ruby-version').readline.chomp
```
Comment (skrix) [inline Gemfile:4]: Too much as for me, we don't change ruby-version so frequently to use this thing, but as you wish ;)

File: config/sidekiq.docker.yml [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+production:
+  :concurrency: 20
+:queues:
+  - default
+  - mailers
+
+:schedule:
+  FetchAllTalksDataJob:
+    every: '1d' # Runs every day
+    class: FetchAllTalksDataJob
```
Comment (skrix) [inline config/sidekiq.docker.yml:13]: add a new line at the end of file 😆

File: docker-compose.deploy.yml [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+      - pivorak-certbot-conf:/etc/letsencrypt
+      - pivorak-cerbot-www:/var/www/certbot
+    ports:
+      - "80:80"
+      - "443:443"
+    networks:
+      - pivorak-network
+    depends_on:
+      - app
+    #command: "tail -f /dev/null"
```
Comment (skrix) [inline docker-compose.deploy.yml]: probably you forget to remove this ;)

File: docker-compose.deploy.yml [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+        parallelism: 1
+        delay: 10s
+        order: start-first
+      restart_policy:
+        condition: on-failure
+    healthcheck:
+      test: ["CMD", "redis-cli", "-h", "localhost", "ping"]
+      timeout: 1s
+      retries: 30
+  nginx:
```
Comment (skrix) [inline docker-compose.deploy.yml:98]: add a newline to separate nginx from redis ;)

File: docker-compose.deploy.yml [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+      DOMAINS: pivorak.com
+      SERVER_CONTAINER_LABEL: nginx.proxy
+      WEBROOT_PATH: /tmp/letsencrypt/www
+      CERTS_PATH: /etc/ssl/certs
+      CHECK_FREQ: 1
+    networks:
+      - pivorak-network
+    entrypoint: "/bin/sh -c 'trap exit TERM; while :; do certbot renew; sleep 12h & wait $${!}; done;'"
+
+
```
Comment (skrix) [inline docker-compose.deploy.yml]: 1 instead 2 ;)

File: docker-compose.deploy.yml [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+    #command: "tail -f /dev/null"
+    command: "/bin/sh -c 'while :; do sleep 6h & wait $${!}; nginx -s reload; done & nginx -g \"daemon off;\"'"
+  certbot:
+    image: certbot/certbot
+    restart: unless-stopped
+    volumes:
+      - pivorak-certbot-conf:/etc/letsencrypt
+      - pivorak-cerbot-www:/var/www/certbot
+    environment:
+      EMAIL: bohdan@varshchuk.com
```
Comment (skrix) [inline docker-compose.deploy.yml]: safe enough for you?)

File: docker-entrypoint.deploy.sh [hunk 15]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+# Interpreter identifier
```
Comment (skrix) [inline docker-entrypoint.deploy.sh]: just a question this file required to live in the root of the project?

File: docker-entrypoint.sh [hunk 16]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (skrix) [inline docker-entrypoint.sh:1]: same as above ;)

File: s6/usr/local/sbin/init_db [hunk 17]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+echo "Creating pivorak_production database\n";
+psql -h postgres -U postgres -tc "SELECT 1 FROM pg_database WHERE datname = 'pivorak_production'" | grep -q 1 || psql -h postgres -U postgres -c "CREATE DATABASE pivorak_production WITH OWNER postgres ENCODING 'UTF8' LC_COLLATE = 'en_US.UTF-8' LC_CTYPE = 'en_US.UTF-8' TEMPLATE template0;"
```
Comment (skrix) [inline s6/usr/local/sbin/init_db:3]: can you please use multiline command?)

File: s6/usr/local/sbin/start_jobs [hunk 18]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+echo "Starting background_jobs\n";
```
Comment (skrix) [inline s6/usr/local/sbin/start_jobs]: Looks like it's not `Starting` but `Restarting ` am I wrong?

File: s6/usr/local/sbin/start_api [hunk 19]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+while :
```
Comment (skrix) [inline s6/usr/local/sbin/start_api:2]: more visibility with pre-echo like in `s6/usr/local/sbin/start_jobs` will be great AFM)

**Review by skrix — COMMENTED** [review]
Really complicated and cool stuff, thx ;)

File: config/database.docker.yml [hunk 20]
```diff
@@ -0,0 +1,19 @@
+default: &default
+  encoding: unicode
+  adapter:  postgresql
+  host:     postgres
+  pool:     20
+  username: <%= ENV['DATABASE_USER'] %>
+  password:
```
Comment (skrix) [inline config/database.docker.yml:7]: Why empty?

File: s6/etc/cont-init.d/06-assets-and-cleanup [hunk 21]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/execlineb -P
+
+with-contenv
+rm -rf /app/public/uploads/tmp
+s6-envuidgid -D 32769:32769 daemon
+# Migrate db
+bundle exec rake assets:precompile --trace
```
Comment (skrix) [inline s6/etc/cont-init.d/06-assets-and-cleanup]: wat? `# Migrate db`?

**Review by skrix — APPROVED** [review]
Small comments and can be merged ;)

File: config/database.docker.yml (hunk 20 above)
Comment (g3d) [reply config/database.docker.yml:7]: Coz by default postgres container does not set password

**Pull request merged**